components/ratgdo/number/ratgdo_number.cpp

```diff
@@ -32,6 +32,16 @@ namespace ratgdo {
                 value = 0;
             }
         }
+        else {
+            if (this->number_type_ == RATGDO_CLIENT_ID) {
+                uint32_t int_value = static_cast<uint32_t>(value);
+                if ((int_value & 0xFFF) != 0x539) {
+                    int_value = (int_value & ~0xFFF) | 0x539;
+                    value = int_value;
+                    this->pref_.save(&value);
```

Comment:
Every one of them ended up being `1337` after reflashing

Reply:
Yeah, that's expected is they were < `0xfff` (4095) before flashing (`1337`=`0x539`).

Reply:
Maybe it makes sense to generate a new one using `value = ((random_uint32() + 1) % 0xFFFF) << 12 | 0x539;` if it gets past  `if ((int_value & 0xFFF) != 0x539) {`?

That should ensure everyone gets a new working id and they don't end up back at 1337 in case the counter is too far behind?

Reply:
Sure.